### PR TITLE
AWS, GCP: Include dependency notices in bundle notice

### PR DESCRIPTION
**Pull request by bryanck**
This PR includes the NOTICE from dependencies in the bundle's NOTICE file. This was generated using the [Gradle License Report](https://github.com/jk1/Gradle-License-Report) plugin with some manual edits following generation.

### CODE REVIEW

File: aws-bundle/NOTICE [hunk 1]
```diff
@@ -4,3 +4,885 @@ Copyright 2017-2023 The Apache Software Foundation
...
+
+--------------------------------------------------------------------------------
+
+NOTICE for Group: commons-logging  Name: commons-logging  Version: 1.2
+
+Apache Commons Logging
+Copyright 2003-2014 The Apache Software Foundation
+
+This product includes software developed at
+The Apache Software Foundation (http://www.apache.org/).
```
Comment (rdblue) [inline aws-bundle/NOTICE:38]: I hate to make this more annoying, but we _don't_ want to include a NOTICE if it contains only ASF boilerplate like this. We can omit it and should be cause we need to produce minimal NOTICE content.

File: aws-bundle/NOTICE [hunk 2]
```diff
@@ -4,3 +4,885 @@ Copyright 2017-2023 The Apache Software Foundation
...
+
+Apache HttpCore
+Copyright 2005-2020 The Apache Software Foundation
+
+This product includes software developed at
+The Apache Software Foundation (http://www.apache.org/).
+
+--------------------------------------------------------------------------------
+
+NOTICE for Group: software.amazon.awssdk  Name: annotations  Version: 2.20.18
```
Comment (rdblue) [inline aws-bundle/NOTICE:62]: Can we deduplicate these and have multiple lines like this for NOTICE blocks that are identical?

Comment (bryanck) [reply aws-bundle/NOTICE]: Sure, this is done.

File: aws-bundle/NOTICE (hunk 1 above)
Comment (bryanck) [reply aws-bundle/NOTICE]: OK, this is done.

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Thanks for fixing this, @bryanck!

**Pull request merged**